scipy/signal/filter_design.py

```diff
@@ -4272,25 +4321,17 @@ def ellipap(N, rp, rs):
         z = []
         return asarray(z), asarray(p), k
 
-    eps = numpy.sqrt(10 ** (0.1 * rp) - 1)
-    ck1 = eps / numpy.sqrt(10 ** (0.1 * rs) - 1)
-    ck1p = numpy.sqrt(1 - ck1 * ck1)
-    if ck1p == 1:
+    eps_sq = 10 ** (0.1 * rp) - 1
+    eps = np.sqrt(eps_sq)
+    ck1_sq = eps_sq / (10 ** (0.1 * rs) - 1)
+    if ck1_sq == 0:
```

Comment:
The version of `ellipap` in this PR fails for rp < 1e-15 (or about there; 1e-16 fails) even if I remove the `rp == 0` check.

For `N==1` this is from a divide-by-zero in the earlier `1 / (10 ** (0.1 * rp) - 1)`; this might be solvable with `expm1`.

For `N==2`, `rp==1e-16` results in `krat == 0`; that is disallowed by `_solve_kratio`.  Now that I look at it again, `_solve_kratio` should probably return 0 for `krat==0`.  However, besides this, `_solve_vratio` further down takes `1 / sqrt(10 ** (0.1 * rp) - 1)` as its argument; again; this is possibly again fixable by using `expm1`.

I can explore allowing small `rp` values via using `expm1` -- would this be useful?  I would have thought for a sufficiently small `rp` value one might as well use the equivalent Chebyshev design functions.

I could change the error message ("if rp is very small, consider using cheby2").

Is it naive to expect `ellip(2, 1e-7, 20, 0.1)` to give a similar result to `cheby2(2, 20, 0.1)` ? They're not at all close; for the version of ellipap of this PR:

```
root@e5a8651e374b:/home/scipy# python3.7
Python 3.7.5 (default, Nov  7 2019, 10:50:52) 
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from scipy.signal import cheby2, ellip
>>> cheby2(2, 20, 0.1, output='zpk')
(array([0.90445128-0.42657694j, 0.90445128+0.42657694j]), array([0.90449059-0.09550526j, 0.90449059+0.09550526j]), 0.09546596712030596)
>>> ellip(2, 1e-10, 20, 0.1, output='zpk')
(array([-0.9999231+0.0124011j, -0.9999231-0.0124011j]), array([-0.97362727-0.02832822j, -0.97362727+0.02832822j]), 0.9740392272955193)
```

Results for e4238172b, from which this PR is based, are similar:

```
root@e5a8651e374b:/home/scipy# python3.7
Python 3.7.5 (default, Nov  7 2019, 10:50:52) 
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from scipy.signal import cheby2, ellip
>>> cheby2(2, 20, 0.1, output='zpk')
(array([0.90445128-0.42657694j, 0.90445128+0.42657694j]), array([0.90449059-0.09550526j, 0.90449059+0.09550526j]), 0.09546596712030596)
>>> ellip(2, 1e-10, 20, 0.1, output='zpk')
(array([-0.99992397+0.01233108j, -0.99992397-0.01233108j]), array([-0.97361247-0.02831258j, -0.97361247+0.02831258j]), 0.9740239771319361)
```